### PR TITLE
Improve code quality 2

### PR DESCRIPTION
**Pull request by iamcarbon**
This PR has no functional changes and is focused on improving code quality / consistency. 

Posting for early feedback.

### CODE REVIEW

File: SgmlReaderDll/SgmlParser.cs [hunk 1]
```diff
@@ -1503,35 +1500,35 @@ public class ElementDecl
...
         /// <summary>
         /// The element name.
         /// </summary>
-        public string Name => m_name;
+        public string Name { get; }
 
         /// <summary>
         /// The <see cref="Sgml.ContentModel"/> of the element declaration.
         /// </summary>
-        public ContentModel ContentModel => m_contentModel;
```
Comment (lovettchris) [inline SgmlReaderDll/SgmlParser.cs:1456]: Why did you change this?  => m_contentModel is the now recommended way of exposing read only properties on a private member variable.

**Review by lovettchris — CHANGES_REQUESTED** [review]
Thanks, I'm generally not a fan of removing "this." and I'm not a huge fan of adopting the latest language 9 features just because.  But you have a lot of other good stuff in here.  But the one open issue on replacing => m_contentModel with {get;} why was that change made?

**Comment by iamcarbon** [comment]
@lovettchris 

Agreed on removing the readonly auto-implemented properties. 

After removing the m_ prefix from the private variables, this felt redundant and out of place. This is likely from having spent a lot of time with the core runtime libraries which standardized on a _ without the this. prefix. Here is a link to the runtime library conventions. Notable are rules 3 (Prefix internal and private instance fields with _) and 4 (We avoid this. unless absolutely necessary.).

https://github.com/dotnet/runtime/blob/10381a2cdc33860f0dc649b24cbb703d23b9ea33/docs/coding-guidelines/coding-style.md

Happy to prefix them again if you prefer. 

The C# 9 pattern matching features do aid in readability, particularly when used in switch expressions. Let me know if you think these pattern uses should be reverted.

File: SgmlReaderDll/SgmlParser.cs (hunk 1 above)
Comment (lovettchris) [reply SgmlReaderDll/SgmlParser.cs:1456]: thanks.

**Comment by lovettchris** [comment]
No need to revert anything, it looks good now.

**Comment by lovettchris** [comment]
Published: https://www.nuget.org/packages/Microsoft.Xml.SgmlReader/1.8.20

**Pull request merged**